### ISSUE TITLE
Issue with parsing rpvars structure in latest 24.1

### ISSUE THREAD

**Issue by mkundu1** [ansys/pyfluent#2174]
```
>>> import ansys.fluent.core as pyfluent
>>> solver = pyfluent.launch_fluent()              
>>> len(solver.rp_vars())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\ANSYSDev\PyFluentDev\pyfluent\src\ansys\fluent\core\rpvars.py", line 59, in __call__
    else (self._get_var(var) if var else self._get_vars())
                                         ^^^^^^^^^^^^^^^^
  File "D:\ANSYSDev\PyFluentDev\pyfluent\src\ansys\fluent\core\rpvars.py", line 87, in _get_vars
    return {val[0]: val[1] for val in list_val}
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\ANSYSDev\PyFluentDev\pyfluent\src\ansys\fluent\core\rpvars.py", line 87, in <dictcomp>
    return {val[0]: val[1] for val in list_val}
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'list'
```
Some real-list type rpvars like `(morpher/y-coords-custom () real-list #f ())` are parsed incorrectly as `(['morpher/y-coords-custom', [], 'real-list', False, []], '#[free')`. I think it is not ideal to parse the rpvars structure from a string representation. The eval/scheme_eval rpcs in scheme_eval service can return a Scheme structure as Python structure, we can use that here.

Related #2141 (found the current issue while investigating this)

**Comment by mkundu1** [comment]
Another thing to consider here is to restrict `solver.rp_vars()` to return only the rpvars which are persisted in a case file. I think `'#[free'` appears due to some rpvars which are not serializable.

**Comment by mkundu1** [comment]
The issue is in the scheme-eval service on Fluent side, will fix in 24.2.

Not changing the parser as the parsed results must be identical with results from parsing case files for which scheme-eval service is not used.

**Comment by mkundu1** [comment]
Fixed in 24.2
